bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieStateManager.java

```diff
@@ -149,6 +160,30 @@ public Number getSample() {
             }
         };
         statsLogger.registerGauge(SERVER_STATUS, serverStatusGauge);
+        this.serverSanityGauge = new Gauge<Number>() {
+            @Override
+            public Number getDefaultValue() {
+                return -1;
+            }
+
+            @Override
+            public Number getSample() {
+                return sanityPassed.get();
+            }
+        };
+        statsLogger.registerGauge(SERVER_SANITY, serverSanityGauge);
+        stateService.scheduleAtFixedRate(() -> {
+            if (forceReadOnly.get()) {
+                sanityPassed.set(1);
+                return;
+            }
+            SanityTestCommand.handleAsync(conf, new SanityTestCommand.SanityFlags()).thenAccept(__ -> {
```

Comment:
Do we need to check whether the bookie is in ReadOnly mode? so that we can skip running the Sanity test.

Reply:
yes, it's checking at line#176

Reply:
Line #176 only checks forceReadOnly, not check the bookie runs into ReadOnly due to the disk being full.

Reply:
i see. sure, let me add readOnly check as well.

Reply:
done. added the read-only check.

Reply:
sure, done.